Fix all linting errors and enable CI linting

ram-from-tvl opened the pull request:
Fixes #297

peterdudfield commented on .github/workflows/pytest_unit.yaml:
```diff
@@ -7,7 +7,27 @@ on:
 
 jobs:
-  branch-ci:
+  linting:
```
please remove this

peterdudfield commented on .github/workflows/pytest_unit.yaml:
```diff
@@ -7,7 +7,27 @@ on:
+      - name: Run ruff formatting check
+        run: ruff format --check .
+
+  tests:
```
you can just turn on `enable_linting` below

peterdudfield commented:
Would you be able to add 
```
pull_request_target:
    types: [opened, edited, reopened]
```
to the CI `pytest_unti.yaml`. This should then allows the CI to run in a PR

peterdudfield commented:
Actually need to remove
```
  pull_request:
    types: [opened, synchronize, reopened]
  schedule:
    - cron: "0 12 * * 1"
  pull_request_target:
    types: [opened, synchronize, reopened, ready_for_review]
``` 
from integration tests, and put this over to `pytest_unti.yaml.` I can do this in a seperate PR if you prefer

ram-from-tvl commented:
> Actually need to remove
> 
> ```
>   pull_request:
>     types: [opened, synchronize, reopened]
>   schedule:
>     - cron: "0 12 * * 1"
>   pull_request_target:
>     types: [opened, synchronize, reopened, ready_for_review]
> ```
> 
> from integration tests, and put this over to `pytest_unti.yaml.` I can do this in a seperate PR if you prefer

Sure @peterdudfield, please let me know if I should raise a separate PR for this after the current one is merged, and whether it would be helpful to open an issue as well.

ram-from-tvl commented:
Hi @peterdudfield 
I have made the requested changes. Please review it and let me know if any changes are required.
Thank you!

peterdudfield commented:
> @peterdudfield I've moved the pull request triggers from pytest.yaml (integration tests) to pytest_unit.yaml as requested. Unit tests will now run on PRs while integration tests only run on push/schedule. Please let me know if its good to go. Thank you!

Thanks for this, we can now see if the CI is passing or not. Its not. 

It looks like we need ruff in the pyproject.toml requirements. You ok to add this?

ram-from-tvl commented:
> > @peterdudfield I've moved the pull request triggers from pytest.yaml (integration tests) to pytest_unit.yaml as requested. Unit tests will now run on PRs while integration tests only run on push/schedule. Please let me know if its good to go. Thank you!
> 
> Thanks for this, we can now see if the CI is passing or not. Its not.
> 
> It looks like we need ruff in the pyproject.toml requirements. You ok to add this?

I have added Ruff @peterdudfield. please let me know if this looks good.
Thank you!

ram-from-tvl commented:
@peterdudfield 
I think all tests are passing. Hope we are good to go. LGTM!
Thank you!
